cardinal/gamestate/map.go

```diff
@@ -0,0 +1,272 @@
+package gamestate
+
+import (
+	"context"
+
+	"github.com/rotisserie/eris"
+)
+
+type MapStorage[K comparable, V any] struct {
```

Comment:
Looks like `MapStorage` is meant to adhere to the `PrimitiveStorage` interface, maybe we should add the following to ensure it:

`var _ MapStorage = &PrimitiveStorage{}`

Reply:
Because both MapStorage and PrimitiveStorage are generic this can't be done. 

It compiles anyway so the compiler recognizes the genericness of it.

Reply:
this works: `var _ PrimitiveStorage[string] = &MapStorage[string, any]{}`